src/highdicom/content.py

```diff
@@ -1478,15 +1503,318 @@ def __init__(
+        if lut_data.dtype.type == np.uint8:
+            bits_per_entry = 8
+        elif lut_data.dtype.type == np.uint16:
+            bits_per_entry = 16
+        else:
+            raise ValueError(
+                "Numpy array must have dtype uint8 or uint16."
+            )
+
+        if color.lower() not in ('red', 'green', 'blue'):
+            raise ValueError(
+                'Argument "color" must be either "red", "green", or "blue".'
+            )
+        self._attr_name_prefix = f'{color.title()}PaletteColorLookupTable'
+
+        setattr(self, f'{self._attr_name_prefix}Data', lut_data.tobytes())
+        setattr(
+            self,
+            f'{self._attr_name_prefix}Descriptor',
+            [len_data, int(first_mapped_value), bits_per_entry]
+        )
+
+    @property
+    def lut_data(self) -> np.ndarray:
+        """numpy.ndarray: lookup table data"""
+        if self.bits_per_entry == 8:
+            dtype = np.uint8
+        elif self.bits_per_entry == 16:
+            dtype = np.uint16
+        else:
+            raise RuntimeError("Invalid LUT descriptor.")
+        length = self.number_of_entries
+        data = getattr(self, f'{self._attr_name_prefix}Data')
+        array = np.frombuffer(data, dtype)
+        if len(array) != length:
+            raise RuntimeError(
+                'Length of Lookup Table Data does not match the value '
+                'expected from the Lookup Table Descriptor. '
+                f'Expected {length}, found {len(array)}.'
+            )
+        return array
+
+    @property
+    def number_of_entries(self) -> int:
+        """int: Number of entries in the lookup table."""
+        descriptor = getattr(self, f'{self._attr_name_prefix}Descriptor')
+        return int(descriptor[0])
+
+    @property
+    def first_mapped_value(self) -> int:
+        """int: Pixel value that will be mapped to the first value in the
+        lookup table.
+        """
+        descriptor = getattr(self, f'{self._attr_name_prefix}Descriptor')
+        return int(descriptor[1])
+
+    @property
+    def bits_per_entry(self) -> int:
+        """int: Bits allocated for the lookup table data. 8 or 16."""
+        descriptor = getattr(self, f'{self._attr_name_prefix}Descriptor')
+        return int(descriptor[2])
+
+
+class SegmentedColorLUT(Dataset):
+
+    """Class that abstracts a segmented pixel value palette color lookup table.
+    """
+
+    def __init__(
+        self,
+        first_mapped_value: int,
+        number_of_entries: int,
+        segmented_lut_data: np.ndarray,
+        color: str
+    ):
+        """
+
+        Parameters
+        ----------
+        first_mapped_value: int
+            Pixel value that will be mapped to the first value in the
+            lookup table.
+        number_of_entries: int
+            Total number of entries in the expanded lookup table
+        segmented_lut_data: numpy.ndarray
+            Segmented lookup table data. Must be of type uint8 or uint16.
```

Comment:
Fully agreed. I was thinking that we could implement an alternative constructor that would create an instance of the class similar to [matplotlib.colors.LinearSegmentedColormap.from_list()](https://matplotlib.org/stable/api/_as_gen/matplotlib.colors.LinearSegmentedColormap.html#matplotlib.colors.LinearSegmentedColormap.from_list). However, the way segemented colormaps are specified in matplotlib differs from the way it's being done in DICOM, and then I thought it may just add further confusion.

Reply:
It's hard to check the validity of `segmented_lut_data` without parsing it in its entirety. Currently, this is done lazily when the `lut_data` property is accessed. We could consider doing this upon construction, but it could take some time. What do you think?

Reply:
I don't think that equidistant colors like the matplotlib function makes sense, but passing `(value, color)` tuples probably would. I think we should do it but not hold up this PR over it, just point to the standard for now.

I think we should parse it upon construction. Most of highdicom's added value is making sure things are correct so I think we should not let though incorrect things here. We wouldn't need to actually build the output array in the constructor which may save some time

Reply:
I agree about the checks. We can try and see how long it takes.

Reply:
Addressed with b71085db106860137b7dbb0fc023eba44c636963